src/libcrun/cgroup-resources.c

```diff
@@ -937,7 +938,17 @@ write_cpu_resources (int dirfd_cpu, bool cgroup2, runtime_spec_schema_config_lin
           len = sprintf (fmt_buf, "%" PRIu64, cpu->period);
           ret = write_cgroup_file (dirfd_cpu, "cpu.cfs_period_us", fmt_buf, len, err);
           if (UNLIKELY (ret < 0))
-            return ret;
+            {
+              if (errno != EINVAL)
+                return ret;
+              /* Sometimes when the period to be set is smaller than the current one,
+               * it is rejected by the kernel (EINVAL) as old_quota/new_period exceeds
+               * the parent cgroup quota limit. If this happens and the quota is going
+               * to be set, ignore the error for now and retry after setting the quota.
+               */
```

Comment:
since the `err` is not used, we'll need to free it with `crun_error_release (err);`

Reply:
fixed